packages/ui/src/components/Pagination/SPagination.vue

```diff
@@ -0,0 +1,377 @@
+     * text shown near sizes dropdown
+     * */
+    sizesLabel?: string
+  }>(),
+  {
+    total: 0,
+    pageSize: null,
+    currentPage: 1,
+    pageSizes: () => [10, 50, 100],
+    sizesLabel: 'Rows per page',
+  },
+)
+/* eslint-disable @typescript-eslint/unified-signatures */
+const emit = defineEmits<{
+  (event: 'size-change', value: number): void
+  (event: 'current-change', value: number): void
+  (event: 'prev-click', value: number): void
+  (event: 'next-click', value: number): void
+  (event: 'update:currentPage', value: number): void
+  (event: 'update:pageSize', value: number): void
+}>()
+
+const current = useVModel(props, 'currentPage', emit, { passive: true })
+const size = useVModel(props, 'pageSize', emit, { passive: true })
+const numericSize = computed({
+  get: () => {
+    return size.value ?? (props.pageSizes[0] || 10)
+  },
+  set: (value: null | number) => {
+    size.value = value
+  },
+})
+
+const sizeOptions = computed(() => props.pageSizes.map((size) => ({ label: String(size), value: size })))
+
+const pagesNum = computed(() => Math.max(1, Math.ceil(props.total / numericSize.value)))
+const isCurrentFirst = computed(() => current.value === 1)
+const isCurrentLast = computed(() => current.value === pagesNum.value)
+const isAbleJumpPrev = computed(() => current.value >= PAGINATION_JUMP_SIZE)
+const isAbleJumpNext = computed(() => current.value <= pagesNum.value - PAGINATION_JUMP_SIZE + 1)
+const shouldShowControls = computed(() => pagesNum.value > PAGINATION_MAX_PAGES_SELECTABLE)
+const firstItemNum = computed(() => (current.value - 1) * numericSize.value + 1)
+const lastItemNum = computed(() => Math.min(props.total, current.value * numericSize.value))
+
+const isJumpPrevButtonHovered = ref(false)
+whenever(not(isAbleJumpPrev), () => {
+  isJumpPrevButtonHovered.value = false
+})
+const isJumpNextButtonHovered = ref(false)
+whenever(not(isAbleJumpNext), () => {
+  isJumpNextButtonHovered.value = false
+})
+
+const pagination = ref(null)
+const paginationContainerWidth = ref(0)
+useResizeObserver(pagination, (entries) => {
+  setTimeout(() => {
+    paginationContainerWidth.value = entries[0].contentRect.width
+  })
+})
+
+const paginationLeftPanel = ref(null)
+const paginationLeftPanelWidth = ref(0)
+useResizeObserver(paginationLeftPanel, (entries) => {
+  setTimeout(() => {
+    paginationLeftPanelWidth.value = entries[0].contentRect.width
+  })
+})
+
+const paginationRightPanel = ref(null)
+const paginationRightPanelWidth = ref(0)
+useResizeObserver(paginationRightPanel, (entries) => {
+  setTimeout(() => {
+    paginationRightPanelWidth.value = entries[0].contentRect.width
+  })
+})
+
+const shouldWrap = ref(false)
+watch([paginationContainerWidth, paginationLeftPanelWidth, paginationRightPanelWidth], () => {
+  shouldWrap.value =
+    paginationContainerWidth.value < paginationLeftPanelWidth.value + paginationRightPanelWidth.value ||
+    paginationContainerWidth.value <= PAGINATION_BREAKPOINT_WIDTH
+})
```

Comment:
Why don't write `shouldWrap` as a `computed`?

Reply:
Ok, thanks. There were another reactive variables, so explicit dependencies list used to make sense